suse/factory.go

```diff
@@ -0,0 +1,236 @@
+package suse
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/Masterminds/semver"
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/pkg/ovalutil"
+	"github.com/quay/zlog"
+	"golang.org/x/net/html"
+)
+
+//doc:url updater
+const base = `https://ftp.suse.com/pub/projects/security/oval/`
+
+var (
+	reELFile   = regexp.MustCompile(`suse.linux.enterprise.server.([1-9][1-9]).xml.gz`)
+	reLeapFile = regexp.MustCompile(`opensuse.leap.(1[0-9].[0-9]+).xml.gz`)
+
+	minimumLEAP = semver.MustParse("15.5")
+)
+
+type Factory struct {
+	c    *http.Client
+	base *url.URL
+
+	mu sync.Mutex
+}
+
+// UpdaterSet implements [driver.UpdaterSetFactory].
+func (f *Factory) UpdaterSet(ctx context.Context) (driver.UpdaterSet, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "suse/Factory.UpdaterSet")
+	us := driver.NewUpdaterSet()
+	if f.c == nil {
+		zlog.Info(ctx).
+			Msg("unconfigured")
+		return us, nil
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, f.base.String(), nil)
+	if err != nil {
+		return us, fmt.Errorf("suse: unable to construct request: %w", err)
+	}
+	f.mu.Lock()
+	defer f.mu.Unlock()
```

Comment:
also just noticed this, sorry. What is this protecting? I'm wondering if we have to lock around the HTTP call or if this can be done later

Reply:
I believe this was ensuring that multiple calls to UpdaterSet were protecting setting the fields belonging to the Factory, but we're no longer saving etag stuff there I think it's ok to remove